src/metatensor/models/utils/data/dataset.py

```diff
@@ -173,3 +209,33 @@ def _train_test_random_split(
     lengths /= lengths.sum()
 
     return random_split(dataset=train_dataset, lengths=lengths, generator=generator)
+
+
+def group_and_join(
```

Comment:
same question as for the dataset?

Reply:
Yes, same answer. Temporary, but not short-term I would say